onnxruntime/core/providers/shared_library/provider_bridge_provider.cc

```diff
@@ -17,9 +17,13 @@
 #define _Post_writable_byte_size_(n)
 #endif
 
+#ifndef _GLIBCXX_USE_NOEXCEPT 
+#define _GLIBCXX_USE_NOEXCEPT 
+#endif
+
 // Override default new/delete so that we match the host's allocator
 _Ret_notnull_ _Post_writable_byte_size_(n) void* operator new(size_t n) { return Provider_GetHost()->HeapAllocate(n); }
-void operator delete(void* p) { return Provider_GetHost()->HeapFree(p); }
+void operator delete(void* p) _GLIBCXX_USE_NOEXCEPT { return Provider_GetHost()->HeapFree(p); }
```

Comment:
Wouldn't just having 'noexcept' work for all compilers? This avoids the macro for just the one compiler.

Reply:
Looks like noexcept is standard since c++ 11 -  @snnn - any concerns?

https://www.cplusplus.com/reference/new/operator%20delete/

Reply:
No. It would be great. Thank you @RyanUnderhill

Reply:
@chandru-r please feel free to use any C++11/14 language features.

Reply:
Fixed.